app/views/bus_stops/_form.haml

```diff
@@ -48,4 +48,7 @@
         %td= f.check_box :completed
     %tr
       %td= submit_tag 'Save stop', class: 'btn btn-primary btn-lg btn-block submit'
+      - if session[:route_id]
```

Comment:
Generally should use instance variables to control view logic whenever possible.